embark/uploader/views.py

```diff
@@ -61,15 +63,25 @@ def upload_file(request):
 
         if form.is_valid():
             logging.info("Posted Form is valid")
-
-            title = form.cleaned_data['version']
-            logging.info(title)
-            # form.save()
-            return HttpResponse("Uploaded")
+            form.save()
+
+            # get relevant data
+            # TODO: make clean db access
+            firmware_file = form.cleaned_data['firmware']
+            firmware_flags = Firmware.objects.latest('id')
+
+            # inject into bounded Executor
+            if boundedExecutor.submit_firmware(firmware_flags=firmware_flags, firmware_file=firmware_file):
+                return HttpResponseRedirect("../../home/#uploader")
+            else:
+                return HttpResponse("queue full")
         else:
             logging.info("Posted Form is unvalid")
+            print(form.errors)
```

Comment:
debug?

Reply:
Can be removed and replaced with logger entry